### PR TITLE
feat(CAD): Add CAD to the signup tab in firstrun

### LINKED ISSUE #4944: Redirect firstrun users to CAD in the signup tab after user verifies their email
Originally reported by @ryanfeeley in https://github.com/mozilla/fxa-content-server/issues/4939.

On the firstrun page, instead of showing the user "Account verified" when we have detected the user verified, we should send the user to "/connect another device" so we have more touch points that encourage users to connect a 2nd device.

### PR DESCRIPTION
**Pull request by shane-tomlinson**
Builds on https://github.com/mozilla/fxa-content-server/pull/4995

Not even close to ready for review.

TODO:
* [x] Sizing of app store buttons. (with existing firstrun iframe size, no need to resize the buttons)
* [x] Resize window for modal, or stop using modals if in the iframe.
* [x] What do we do for "Maybe later" on the /sms page? User should already be signed in, so sending them to "/signin" doesn't make sense. Sending them to `/connect_another_device` seems equally strange, we just prompt the user to install Fx for mobile, which the /sms screen does. (decided to remove)
* [x] Update lots of functional tests.
* [x] In metrics, figure out how to differentiate between users who perform an action from the signup and verification tabs.

fixes #4944

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
Copied from #4944 

-------------------

@ryanfeeley - Thanks for the help with the "why is this required" modal, I'm going to put a bit of work into getting the iframe to resize for the modal text. We talked about updating the text to be a bit clearer, did you have text in mind? (@davismtl made a suggestion, see below)

screen shot 2017-05-02 at 16 11 33

Now for the main sms form - here's a screenshot within the iframe:

screen shot 2017-05-02 at 16 00 31

Think the app store buttons are too big?


-------------------------

From [@davismtl's comment about the modal text](https://github.com/mozilla/fxa-content-server/issues/4944#issuecomment-297788953):

> As per the copy, I really don't like that first sentence. I'm not fond of the parentheses in the headline either. I had to read it a few times. Example of shortening the first 2 sentences into 1:
>
> "Sync lets you securely find your bookmarks, passwords and more from any of your devices running Firefox."

**Comment by ryanfeeley** [comment]
@shane-tomlinson Addressing play/app store badges first, Apple states [1] that minimum size"40 pixels for use onscreen" and Google states [2] that their badge "should be the same size or larger than other application store badges". Appears as though we can put them as close as 10 pixels apart.

Does this give us any wiggle room in responsive mode?

[1] https://developer.apple.com/app-store/marketing/guidelines/#downloadOnAppstore
[2] https://play.google.com/intl/en_us/badges/

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: c9ca9db71ffef936063c1938fd8a4a2ede77a06d
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: c9ca9db71ffef936063c1938fd8a4a2ede77a06d
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

File: app/scripts/views/base.js [hunk 1]
```diff
@@ -936,29 +934,27 @@ define(function (require, exports, module) {
...
      * @returns {Promise}
      */
-    invokeBrokerMethod (methodName/*, ...*/) {
-      var args = [].slice.call(arguments, 1);
-
-      var broker = this.broker;
-
-      return p(broker[methodName].apply(broker, args))
-        .then(this.invokeBehavior.bind(this));
+    invokeBrokerMethod (methodName, ...args) {
```
Comment (shane-tomlinson) [inline app/scripts/views/base.js]: Put these changes into a separate PR.

**Comment by shane-tomlinson** [comment]
This code is running on https://stomlinson.dev.lcip.org 
The easiest way to test is use the mock firstrun page @ http://github.com/shane-tomlinson/firstrun-example

* Start that server
* with fxa-local-dev, type `FXA_ENV=stomlinson npm start`
* visit http://127.0.0.1:8111?endpoint=stomlinson
* Sign up, verify different browser or different tab in same browser.
* Go back to the signup tab, the default is to go to SMS page, to see connect_another_device, load http://127.0.0.1:8111?endpoint=stomlinson&forceExperimentGroup=control
* Continue flow to see if it's good.

File: tests/functional/fx_firstrun_v1_sign_up.js [hunk 2]
```diff
@@ -14,6 +14,11 @@ define([
   var email;
   const PASSWORD = '12345678';
 
+  const SELECTOR_CONFIRM_HEADER = '#fxa-confirm-header';
+  const SELECTOR_CONNECT_ANOTHER_DEVICE_HEADER = '#fxa-connect-another-device-header';
+  const SELECTOR_SIGN_UP_HEADER = '#fxa-signup-header';
+  const SELECTOR_SIGN_UP_SUB_HEADER = '#fxa-signup-header .service';
```
Comment (philbooth) [inline tests/functional/fx_firstrun_v1_sign_up.js]: Given that we define these selectors in more than one test module, maybe it will save us future maintenance burden to stick them in a dedicated constants module and then have one place to update them if they change in the future?

Comment (philbooth) [inline tests/functional/fx_firstrun_v1_sign_up.js]: (in a future PR is fine I mean, no need to in this one)

**Comment by philbooth** [comment]
Everything looks good to me, code-wise.

When I try to follow your instructions, @shane-tomlinson, at this step:

> ...to see connect_another_device, load http://127.0.0.1:8111?endpoint=stomlinson&forceExperimentGroup=control

...if I load the quoted URL I just go to the settings view. Am I doing it wrong or is that what I'm supposed to see?

**Comment by shane-tomlinson** [comment]
> ...if I load the quoted URL I just go to the settings view. Am I doing it wrong or is that what I'm supposed to see?

You have to sign out, delete the account, or clear localStorage or else it'll keep sending you to settings.

File: tests/functional/fx_firstrun_v1_sign_up.js (hunk 2 above)
Comment (shane-tomlinson) [inline tests/functional/fx_firstrun_v1_sign_up.js]: Yeah, I've been thinking that too.

**Comment by philbooth** [comment]
I am officially r+.

**Comment by shane-tomlinson** [comment]
@ryanfeeley gave this a 👍 over Vidyo.
